app/controllers/menus_controller.rb

```diff
@@ -7,14 +7,20 @@ def index
   end
 
   def new
-    @menu = Menu.new
+    if current_user && current_user.role == 'owner' && !current_user.restaurant.nil?
```

Comment:
Same here. CanCanCan can be used for some of this functionality, like `current_user.role == 'owner'`
Also, we do have `current_user.owner?` method that could be used here.